src/console_cmds.cpp

```diff
@@ -1271,7 +1320,14 @@ static bool ConEchoC([[maybe_unused]] uint8_t argc, [[maybe_unused]] char *argv[
 	}
 
 	if (argc < 3) return false;
-	IConsolePrint((TextColour)Clamp(atoi(argv[1]), TC_BEGIN, TC_END - 1), argv[2]);
+
+	auto colour = ParseInteger(argv[1]);
+	if (!colour.has_value() || !IsInsideMM(*colour, TC_BEGIN, TC_END - 1)) {
```

Comment:
```suggestion
	if (!colour.has_value() || !IsInsideMM(*colour, TC_BEGIN, TC_END)) {
```
`IsInsideMM` tests [begin, end).